go/test/endtoend/multiorch_helpers.go

```diff
@@ -0,0 +1,632 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endtoend
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/multigres/multigres/go/clustermetadata/topo"
+	"github.com/multigres/multigres/go/clustermetadata/topo/etcdtopo"
+	"github.com/multigres/multigres/go/common/rpcclient"
+	"github.com/multigres/multigres/go/provisioner/local/pgbackrest"
+	"github.com/multigres/multigres/go/test/utils"
+
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
+	multipoolermanagerdatapb "github.com/multigres/multigres/go/pb/multipoolermanagerdata"
+)
+
+// nodeInstance represents a multipooler node for bootstrap testing
+type nodeInstance struct {
+	name           string
+	cell           string
+	grpcPort       int
+	pgPort         int
+	pgctldGrpcPort int
+	dataDir        string
+	pgctldProcess  *exec.Cmd
+	multipoolerCmd *exec.Cmd
+}
+
+// testEnvConfig holds configuration options for test environment setup
+type testEnvConfig struct {
+	tempDirPrefix    string
+	cellName         string
+	database         string
+	shardID          string
+	tableGroup       string
+	durabilityPolicy string
+	stanzaName       string
+}
+
+// testEnv manages the test environment including etcd, topology server, and nodes
+type testEnv struct {
+	t              *testing.T
+	config         testEnvConfig
+	tempDir        string
+	etcdClientAddr string
+	ts             topo.Store
+	backupLocation string
+	nodes          []*nodeInstance
+	multiOrchCmd   *exec.Cmd
+}
+
+// setupMultiOrchTestEnv creates a test environment with etcd and topology server
+// for multiorch integration tests.
+func setupMultiOrchTestEnv(t *testing.T, config testEnvConfig) *testEnv {
+	t.Helper()
+
+	ctx := t.Context()
+
+	// Setup test directory - use /tmp to avoid long paths that exceed Unix socket limits (103 bytes on macOS)
+	tempDir, err := os.MkdirTemp("/tmp", config.tempDirPrefix)
```

Comment:
FYI: At some point I want to make a general helper for making temp directories with short prefixes so we don't have to keep copying this code